pandas/io/formats/printing.py

```diff
@@ -289,7 +289,7 @@ def default_pprint(thing: Any, max_seq_items: int | None = None) -> str:
 
 
 def format_object_summary(
-    obj,
+    obj: Any,
```

Comment:
The doc-string says "must be iterable and support __getitem__" and that is actually not correct - it also needs to support `len`.

How about using `typing.Sequence`? Might need to add more to the union if you get mypy errors (for example pandas._typing.SequenceNotStr and array-like as they are technically not sequences).